shared-helpers/src/views/multiselectQuestions.tsx

```diff
@@ -314,47 +298,121 @@ export const getCheckboxOption = (
-  const options: ApplicationMultiselectQuestionOption[] = []
-
-  if (value) {
-    options.push({
-      key: value,
-      checked: true,
-      extraData: [],
-    })
-  }
+// Get an individual question option checkbox field
+export const getCheckboxOption = (
+  option: MultiselectOption,
+  question: MultiselectQuestion,
+  applicationSection: MultiselectQuestionsApplicationSectionEnum,
+  register: UseFormMethods["register"],
+  setValue: UseFormMethods["setValue"],
+  getValues: UseFormMethods["getValues"],
+  allOptions: string[],
+  watchFields: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [x: string]: any
+  },
+  errors?: UseFormMethods["errors"],
+  trigger?: UseFormMethods["trigger"],
+  exclusiveKeys?: string[]
+) => {
+  const optionFieldName = fieldName(question.text, applicationSection, option.text)
+  const checkboxField = getCheckboxField(
+    option,
+    question,
+    applicationSection,
+    register,
+    setValue,
+    getValues,
+    allOptions,
+    optionFieldName,
+    trigger,
+    exclusiveKeys
+  )
 
-  question?.options?.forEach((option) => {
-    if (option.text !== value) {
-      options.push({
-        key: option.text,
-        checked: false,
-        extraData: [],
-      })
-    }
-  })
+  return multiselectOptionWrapper(
+    checkboxField,
+    option,
+    question,
+    applicationSection,
+    register,
+    watchFields,
+    errors
+  )
+}
 
-  return {
-    multiselectQuestionId: question.id,
-    key,
-    claimed: Object.keys(data)?.length !== 0,
-    options,
-  }
+// Get an individual question option radio field
+export const getRadioOption = (
+  option: MultiselectOption,
+  question: MultiselectQuestion,
+  applicationSection: MultiselectQuestionsApplicationSectionEnum,
+  register: UseFormMethods["register"],
+  setValue: UseFormMethods["setValue"],
+  allOptions: string[],
+  watchFields: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [x: string]: any
+  },
+  getValues: UseFormMethods["getValues"],
+  errors?: UseFormMethods["errors"],
+  trigger?: UseFormMethods["trigger"]
+) => {
+  const optionFieldName = fieldName(question.text, applicationSection, option.text)
+  const radioField = getRadioField(
+    option,
+    register,
+    setValue,
+    allOptions,
+    optionFieldName,
+    getValues,
+    trigger
+  )
+
+  return multiselectOptionWrapper(
+    radioField,
+    option,
+    question,
+    applicationSection,
+    register,
+    watchFields,
+    errors
+  )
 }
 
+function cleanRadioObject(obj: Record<string, any>) {
+  // Remove nulls
+  let cleanedObj = Object.entries(obj).reduce((acc, [key, value]) => {
+    if (value !== null) {
+      acc[key] = value
+    }
+    return acc
+  }, {})
+
+  // Convert "true" to true
+  cleanedObj = Object.entries(cleanedObj).reduce((acc, [key, value]) => {
+    if (
+      value === "true" &&
+      !(key.includes(AddressHolder.Name) || key.includes(AddressHolder.Relationship))
```

Comment:
Why do we need to exclude these two keys? Is it so we don't overwrite somebody who writes "true" into the field?

Reply:
I believe so, not quite sure, but i remember that was the problem, so it's probably it 😄